pkg/api/v1/atlasdeployment_types.go

```diff
@@ -205,9 +214,19 @@ type ServerlessSpec struct {
 	// +kubebuilder:validation:Pattern:=^[a-zA-Z0-9][a-zA-Z0-9-]*$
 	Name string `json:"name"`
 	// Configuration for the provisioned hosts on which MongoDB runs. The available options are specific to the cloud service provider.
-	ProviderSettings *ProviderSettingsSpec `json:"providerSettings"`
-
+	ProviderSettings *ProviderSettingsSpec       `json:"providerSettings"`
 	PrivateEndpoints []ServerlessPrivateEndpoint `json:"privateEndpoints,omitempty"`
+	// Key-value pairs for resource tagging.
+	// +kubebuilder:validation:MaxItems=50
+	// +optional
+	Tags []*TagSpec `json:"tags,omitempty"`
+}
+
+// ServerlessToAtlas converts the ServerlessSpec to native Atlas client Cluster format.
+func (s *ServerlessSpec) ServerlessToAtlas() (*mongodbatlas.Cluster, error) {
+	result := &mongodbatlas.Cluster{}
+	err := compat.JSONCopy(result, s)
+	return result, err
```

Comment:
nitpick, I'd rather be explicit here, even if it si verbose to map each key to its same or similar key value on the original. This way we see what fields we are talking about and, if we ever need to make some non-standard mapping going forward, the change is smaller.

Not a blocker at all

Reply:
BTW, the explicit mapping would not fail, would it? no err needed

Reply:
I went with this method since a similar method is already used elsewhere in the reconciler (`ToAtlas`) If you think the approach you suggested is better, I can look into it! I'm just not entirely sure I'm understanding the benefit of using explicit mapping over what's already there. Would you be able to explain cases we may need non-standard mapping?

Reply:
I noticed this is kind of a standard in the code later on. Please feel free to leave it as is.

Reply:
Why not simply `ToAtlas()` here? seems it is the default name on most other situations, and being a method os serverless already the prefix feels a bit redundant.

Reply:
It feels a bit redundant, yes, but Advanced and Serverless clusters have different structs (`mongodbatlas.AdvancedCluster` and `mongodbatlas.Cluster`) so two methods are needed to return the appropriate struct type